public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts

```diff
@@ -534,3 +547,38 @@ function registerPanelInteractionsReporter(scene: DashboardScene) {
     }
   });
 }
+
+const convertSnapshotData = (snapshotData: DataFrameDTO[]): DataFrameJSON[] => {
+  return snapshotData.map((data) => {
+    return {
+      data: {
+        values: data.fields.map((field) => field.values).filter((values): values is unknown[] => values !== undefined),
+      },
+      schema: {
+        fields: data.fields.map((field) => ({
+          name: field.name,
+          type: field.type,
+          config: field.config,
+        })),
+      },
+    };
+  });
+};
+
+// override panel datasource and targets with snapshot data using the Grafana datasource
+const convertOldSnapshotToScenesSnapshot = (panel: PanelModel) => {
+  // only old snapshots created with old dashboards contains snapshotData
+  if (panel.snapshotData) {
+    panel.datasource = GRAFANA_DATASOURCE_REF;
+    panel.targets = [
+      {
+        refId: panel.snapshotData[0]?.refId ?? '',
+        datasource: panel.datasource,
+        queryType: 'snapshot',
+        // @ts-ignore
```

Comment:
Why the ignore here? (out of curiosity)

Reply:
It's because the type of targets is https://github.com/grafana/grafana/blob/6ce0efeb41e7c9c4392f439a9b10e23486e2a094/public/app/features/dashboard/state/PanelModel.ts#L172 and does not contain the snapshot property. 
Would it be better to do something different here?

Reply:
should it be added as a prop?